calc/lib/helpers.ex

```diff
@@ -58,20 +55,19 @@ defmodule Helpers do
 
   defp convert_to_float(var) do
     case Float.parse(String.trim(var)) do
-      {var, _} -> {:ok, var}
+      {var, ""} -> {:ok, var}
+      {var, _trailing_garbage} -> {:error, "trailing garbage in ‹#{var}›"}
       :error -> {:error, convert_error_message(var)}
     end
   end
 
   defp convert_to_atom(operator) do
-    try do
-      {:ok, String.to_existing_atom(operator)}
-    rescue ArgumentError ->
-      {:error, convert_error_message(operator)}
-    end
+    {:ok, String.to_existing_atom(operator)}
+  rescue ArgumentError ->
+    {:error, convert_error_message(operator)}
   end
 
   defp convert_error_message(var) do
-    ["Cannot convert '" <> inspect(var) <> "' to integer\n"]
+    ["Cannot convert '" <> inspect(var) <> "' to float\n"]
```

Comment:
```suggestion
    "Unexpected input: '" <> inspect(var) <> "'"
```

Вы же эту функцию и из `convert_to_atom/1` вызываете тоже. И я недопонял, зачем возвращать список? Чем строка плоха?